sycl/test/basic_tests/vec_convert.cpp

```diff
@@ -0,0 +1,199 @@
+    return "rtp";
+  case rounding_mode::rtn:
+    return "rtn";
+  default:
+    return "unknown";
+  }
+}
+#endif
+
+template <> struct helper<0> {
+  template <typename T, int NumElements>
+  static void compare(const vec<T, NumElements> &x,
+                      const vec<T, NumElements> &y) {
+    const T xs = x.template swizzle<0>();
+    const T ys = y.template swizzle<0>();
+#ifdef OUTPUT
+    std::cout << 0 << ": given " << xs << " expected " << ys << std::endl
+              << std::endl;
+#endif
+    assert(xs == ys);
+  }
+};
+
+template <int N> struct helper {
+  template <typename T, int NumElements>
+  static void compare(const vec<T, NumElements> &x,
+                      const vec<T, NumElements> &y) {
+    const T xs = x.template swizzle<N>();
+    const T ys = y.template swizzle<N>();
+#ifdef OUTPUT
+    std::cout << N << ": given " << xs << " expected " << ys << std::endl;
+#endif
+    helper<N - 1>::compare(x, y);
+    assert(xs == ys);
+  }
+};
+
+template <typename T, typename convertT, int NumElements,
+          rounding_mode roundingMode>
+void test(const vec<T, NumElements> &ToConvert,
+          const vec<convertT, NumElements> &Expected) {
+#ifdef OUTPUT
+  std::cout << "call convert from " << typeid(T).name() << " to "
+            << typeid(convertT).name() << " for vector of " << NumElements
+            << " with rounding mode " << getRoundingMode<roundingMode>()
+            << std::endl;
+#endif
+  vec<convertT, NumElements> Converted{0};
+  {
+    buffer<vec<convertT, NumElements>, 1> Buffer{&Converted, range<1>{1}};
+    queue Queue;
+    Queue.submit([&](handler &CGH) {
+      accessor<vec<convertT, NumElements>, 1, access::mode::write> Accessor(
+          Buffer, CGH);
+        CGH.single_task<class kernel_name<T, convertT, static_cast<int>(roundingMode)>>([=]() {
+          Accessor[0] = ToConvert.template convert<convertT, roundingMode>();
+        });
+    });
+  }
+  helper<NumElements - 1>::compare(Converted, Expected);
+}
+
+template <typename T, typename R>
+using EnableIfFloatToInt =
+    detail::enable_if_t<!std::is_same<T, R>::value &&
+                            (std::is_floating_point<T>::value ||
+                             std::is_same<T, half>::value) &&
+                            std::is_integral<R>::value,
+                        R>;
+
+template <typename T, typename R>
+using EnableIfNoFloatToInt =
+    detail::enable_if_t<!std::is_same<T, R>::value &&
+                            !((std::is_floating_point<T>::value ||
+                               std::is_same<T, half>::value) &&
+                              std::is_integral<R>::value),
+                        R>;
+
+int main() {
+  test<int, int, 8, rounding_mode::automatic>(
+      sycl::int8{2, 3, 3, -2, -3, -3, 0, 0},
+      sycl::int8{2, 3, 3, -2, -3, -3, 0, 0});
+  test<float, int, 8, rounding_mode::automatic>(
+      sycl::float8{+2.3f, +2.5f, +2.7f, -2.3f, -2.5f, -2.7f, 0.f, 0.f},
+#ifdef ROUNDING_MODES_IMPLEMENTED
```

Comment:
Please, remove this macro.
Rounding modes must be supported by `convert` method.

Reply:
Done.